### PR TITLE
Add Responsible End Of Life & Archiving Policies

### LINKED ISSUE #1185: Revisit Emeritus Stage Definitions
The Sovereign Tech Fund infrastructure project includes supporting sunsetting/emeritus projects.

As we look to audit our projects to see which (if any) might be emeritus, it's a good time to revisit the [current definition](https://github.com/openjs-foundation/cross-project-council/blob/main/PROJECT_PROGRESSION.md#emeritus-stage) and see if we want to make any changes. 

From my perspective, I am hoping we could get a more precise definition of "active development" and the process for nominating and voting on a potential emeritus project. I think we should also look at the sunsetting checklist (per Tobie's suggestion) and think about any additional or changes to the infrastructure piece.

A potential example can be seen [here](https://www.opendaylight.org/about/lifecycle-releases). Specifically section 2.3.5:

> 2.3.5 Termination Review
> - Termination Proposal Posted for 2 weeks:
> - States reason for project termination being sought
> - Calls out impact on other project, users, communities, and how those will be mitigated
> - Indicates where the project would be archived
> - Can be initiated by vote of project committers
> - Can be initiated by TSC or PMC of containing project if there are either no remaining committers for the project or there have been no commits to the SCM in 18 months

### PR DESCRIPTION
**Pull request by bensternthal**
This PR:

1. Refactors Emeritus stage into End Of Life & Archived
2. Details Annual Review Process

Much thanks to @rginn @PaulaPaul @UlisesGascon and others that contributed to this work. 

Fixes #1185

### CODE REVIEW

**Review by ljharb — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

File: PROJECT_PROGRESSION.md
Comment (ovflowd) [inline PROJECT_PROGRESSION.md:225]: For that version? Wouldn't it be for any version?

File: PROJECT_PROGRESSION.md
Comment (ovflowd) [inline PROJECT_PROGRESSION.md:233]: Do we have guidelines of what is an "expected" condition of CPC attempting to reach out maintainers? Or is this just based on good judgement?

File: PROJECT_PROGRESSION.md
Comment (ovflowd) [inline PROJECT_PROGRESSION.md:238]: We should add a EOL vs Archived example/explanation

File: PROJECT_PROGRESSION.md
Comment (ovflowd) [inline PROJECT_PROGRESSION.md:262]: Should this be moved to a dedicated file as we have for incubating/onboarding checklist?

**Review by ovflowd — APPROVED** [review]
LGTM! Left a few comments

File: PROJECT_PROGRESSION.md
Comment (eemeli) [inline PROJECT_PROGRESSION.md:227]: I am confused by this definition. The first paragraph says that EoL projects "will no longer provide support, updates, patches, or enhancements". But the second paragraph says that EoL projects are expected to have a milestone or date after which they'll be Archived, and _then_ "no longer receive support or updates".

Which one is it?

I think one core issue here may be the choice of terms, as "End of life" sounds really rather synonymous with "Archived" in this context. Could we not reconsider something like the current "Emeritus" term instead, which invokes retirement rather than death? Or maybe "Maintenance"?

From the further text, I get the sense that there's an expectation for this EoL phase to be a sort of reverse incubation to being "Archived", and for this process to be a somewhat active and directed one. So what's the plan for current emeritus projects like Sizzle and jQuery UI that have been [steadily](https://github.com/jquery/jquery-ui/commits/main/) receiving maintenance-y commits across the years? If/as they're being maintained, should they get bumped back up to "at-large"?

**Review by tobie — DISMISSED** [review]
Thanks for getting this started! This is a great step forward. I think we should take the opportunity to clean-up the language and be careful not to introduce new notions. For example, there's now _onboarding_, _incubation_, and _graduation_ that kind of describe overlapping phases.

I'm going to reiterate my request to include a state diagram in this doc so that the whole process  and lifecycle is easier to follow for everyone involved, e.g.:

```mermaid

stateDiagram-v2
    direction LR
    [*] --> Application
    Application --> Incubating
    Application --> [*]
    Incubating --> [*]
    Incubating --> Active
    state Active {
        AL : At Large
        Impact --> AL
        AL --> Impact
    }
    Active --> EoL
    EoL --> Active
    EoL --> Archive

```

It would also be good if the different states where presented in order, and _active_ was defined (as either _At Large_ or _Impact_), as its used in the bylaws.

Finally, I think I agree with @ovflowd suggestion to split the checklists out. Should we merge back the project-status repo into this one while we're at it or do we want to continue keeping it as a separate repo?

**Comment by bensternthal** [comment]
@tobie Hmm we did not introduce new notions (at least I do not think so): onboarding, incubation, and graduation exist in the current doc and we did not adjust those as part of this work.  Maybe I am not understanding. 

If we do want to change those stages, I'd suggest doing that in a different PR, I'd like to keep this one atomic to updating Emeritus.

+1 to a wireframe, I do love a good wireframe

+1 to moving the checklists out, I'd prefer to have those in this repo but can defer to the community.

**Comment by tobie** [comment]
> @tobie Hmm we did not introduce new notions (at least I do not think so): onboarding, incubation, and graduation exist in the current doc and we did not adjust those as part of this work. Maybe I am not understanding. 
> If we do want to change those stages, I'd suggest doing that in a different PR, I'd like to keep this one atomic to updating Emeritus.

Oof. My bad. The way the PR is formatted made it look this way. You're right. Entirely fine to move this to a separate PR.

Can we add a diagram to this one and make sure we agree on fundamentals of these new stages? I share @eemeli's confusion about the shapes of the different stages (as I had already mentioned in the original Google doc), and remain unsure whether they refer to periods or discrete events (milestones).

**Comment by tobie** [comment]
> +1 to moving the checklists out, I'd prefer to have those in this repo but can defer to the community.

If you're suggesting we merge back `project-status` into this repo, I'm all for it. We should absolutely do that imho.

**Comment by tobie** [comment]
I remain confused with the different stages.

I think there are two problems:

1. "End of Life" is usually understood as a milestone (see for example how Node talks about it: https://github.com/nodejs/release#release-schedule), but it's being used as a period of time here. I think this is confusing. We should adopt the term "maintenance" instead.
2. There's confusion as to what exactly the EoL (or maintenance) period entails as it says both that the software receives and doesn't receive patches during that time. Here's how I think this would make sense:

| Stage | Feature development | Security patching |
|-------|----------------------|--------------------|
| Active (At Large/Impact) | ✅ | ✅ |
| Maintenance (currently called EoL) | ❌ | ✅ |
| Archived | ❌ |  ❌ |

**Comment by ljharb** [comment]
in that case, i'd call "Archived" "EOL" instead, since "archived" is just a repo status, but EOL is the project status.

File: PROJECT_PROGRESSION.md
Comment (bensternthal) [reply PROJECT_PROGRESSION.md:225]: Agreed.

File: PROJECT_PROGRESSION.md
Comment (bensternthal) [reply PROJECT_PROGRESSION.md:233]: I'd say good judgement for this one. I

File: PROJECT_PROGRESSION.md
Comment (bensternthal) [reply PROJECT_PROGRESSION.md:227]: This is a good point. I will reword to make this clearer. 

I think if projects are in maintenance mode and that they are regularly receiving security updates they should be bumped to active. To me, this is active development unless a date has been communicated when the development would stop. 

End Of Life is meant to communicate clearly. Hey, we are maintaining this now, but at a specified point in the future, we are going to stop.

**Comment by bensternthal** [comment]
Added the wireframe and made some tweaks based on suggestions. Regarding the names for phases, I suggest we keep End Of Life and Archived (see updates) for the following reasons:

1. Maintenance does not imply or communicate a date when maintenance would end, if projects are receiving ongoing security updates and there is no plan to cease updates, then I would say this project is Active.
2. End Of Life projects are expected to have a date when updates would cease. I think it's important to have a stage that clearly communicates, "we are making updates now but we wont after x date."
3. Archived I think is very clear in communicating, you can see the code but no one is working on it. EOL to me still implies folks are working on it and that at some point in the future, they won't be.

**Review by eemeli — CHANGES_REQUESTED** [review]
> Added the wireframe and made some tweaks based on suggestions. Regarding the names for phases, I suggest we keep End Of Life and Archived (see updates) for the following reasons:
> 
> 1. Maintenance does not imply or communicate a date when maintenance would end, if projects are receiving ongoing security updates and there is no plan to cease updates, then I would say this project is Active.

I do not think this matches the practical reality of jQuery UI and Sizzle, two of our five emeritus projects. They are [quite explicitly](https://blog.jqueryui.com/2021/10/jquery-maintainers-update-and-transition-jquery-ui-as-part-of-overall-modernization-efforts/) in "maintenance only" mode, and have not announced any set end date for that. I do not think calling these projects "active" is useful, and splitting up our current project categorization in such a way that none is appropriate for them is not a good change.

As I'm not a contributor myself to these, input from @mgol, @timmywil and/or @Krinkle would be useful here.

> 2. End Of Life projects are expected to have a date when updates would cease. I think it's important to have a stage that clearly communicates, "we are making updates now but we wont after x date."

Why? What is the benefit of communicating such a decision through foundation staging? And why must an end date be included in a decision for a project to go "maintenance only"?

**Comment by timmywil** [comment]
Indeed, UI and Sizzle do not fit either description, and we do not plan on setting an end date. That depends on usage, which we can't really know right now. I consider "maintenance only" to mean that they do receive some important fixes (although worth noting not all of them are security realted), but I'd say the projects are also far from "Active" and not really EOL either.

**Comment by bensternthal** [comment]
Interesting! looping in @PaulaPaul and @UlisesGascon for their thoughts (we worked together on the initial revision).

**Comment by mgol** [comment]
Adding to what Timmy wrote: allowing only security fixes is a pretty limiting factor. In the jQuery UI case, even if we limited current changes as much as possible, we'd still want to maintain support for new jQuery releases which is a jQuery UI dependency and that goes beyond security fixes.

But I would definitely not call jQuery UI support "active". Most feature requests are declined, bugs that existed before the 1.12.x version (the current one is 1.13.2) are generally left for the community to figure out and mostly not acted upon, etc.

**Comment by tobie** [comment]
Agree with the points mentioned above. We absolutely need a stage that says: “This project is complete. If there’s a security issue, it’ll be fixed, but we won’t be adding any new features.”

**Comment by eemeli** [comment]
Agreed with @tobie. Also, "Complete" would be a really good stage name.

**Comment by tobie** [comment]
I like “complete” (or even “feature complete”) as a stage name. It changes the framing quite a bit.

**Comment by tobie** [comment]
"Complete/Feature-complete" stage is explicitly reversible.

We can simplify your table like so:

| Stage | New feature development | Vulnerability & bug fixes |
|-------|---------------------|--------------|
| Active (At Large/Impact) | ✅ | ✅ | 
| "Complete/Feature-complete" (currently called EoL) | ❌ | ✅ |
| "Archived/EoL" | ❌ | ❌ |

File: PROJECT_PROGRESSION.md
Comment (anonrig) [inline PROJECT_PROGRESSION.md:240]:
```suggestion
1. Consensus within private mailing list to move into incubation process
```

**Review by anonrig — APPROVED** [review]

File: PROJECT_PROGRESSION.md
Comment (bensternthal) [reply PROJECT_PROGRESSION.md]: Reworked based on comments in this thread.

**Comment by bensternthal** [comment]
Folks, I did my best to integrate everyone's suggestions and feedback. I think if @tobie and @eemeli approve, we can merge.

**Review by eemeli — APPROVED** [review]

**Review by timmywil — APPROVED** [review]

**Comment by Krinkle** [comment]
I believe these two suggestions were neither applied nor explicitly dismissed:

1. Allow new projects to apply to join in the Complete state.

Perhaps this would fit under the `Incubating` section where we currently say "Projects in this phase may be looking to join the foundation as At-Large or Impact Stage." which could instead read "Projects in this phase may be looking to join the foundation as At-Large, Impact, **or Complete** Stage." 

2. Allow projects to go from Complete back to Active.

I would propose for the text to describe the same criteria in both directions, e.g. maintainer and CPC approval. With the chart also showing an arrow in both directions.

Both of these seem valuable to have in my opinion, but I'm open to understanding why we might prefer not to apply these.

**Review by UlisesGascon — APPROVED** [review]

**Comment by bensternthal** [comment]
@Krinkle I think I have addressed you concerns/comments.

**Review by ljharb — APPROVED** [review]

**Review by Krinkle — APPROVED** [review]

**Pull request merged**
